org.jcryptool.visual.rabin/src/org/jcryptool/visual/rabin/HandleCryptosystemTextbook.java

```diff
@@ -0,0 +1,974 @@
+				boolean checkLength = cstb.textSelector.getText().getText().length() % guiHandler.blocklength == 0;
+				
+				if(!checkLength) {
+					String strLengthCipher = Messages.HandleFirstTab_strLengthCipher;
+					cstb.txtLoadTextWarning.setText(MessageFormat.format(
+							strLengthCipher,
+							(guiHandler.blocklength / 2)));
+					showControl(cstb.txtLoadTextWarning);
+					return;
+				}
+				
+			}
+			else
+				return;
+			
+			String newTextselectorText = null;
+			/*if(firstTextselectorAccessDecryptionMode) {
+				oldTextselectorTextDecryptionMode = cstb.textSelector.getText().getText();
+				firstTextselectorAccessDecryptionMode = true;
+				return;
+			}*/
+			
+			newTextselectorText = cstb.textSelector.getText().getText();
+			if(newTextselectorText != oldTextselectorTextDecryptionMode) {
+				cstb.txtCiphertextSegments.setText("");
+				cstb.cmbChooseBlock.removeAll();
+				cstb.txtChosenPlaintexts.setText("");
+				cstb.txtFirstPlaintext.setText("");
+				cstb.txtSecondPlaintext.setText("");
+				cstb.txtThirdPlaintext.setText("");
+				cstb.txtFourthPlaintext.setText("");
+			}
+			
+			
+			
+			if(!cstb.rftc.txtModN.getText().isEmpty() && !cstb.textSelector.getText().getText().isEmpty()) {
+				if(newTextselectorText != oldTextselectorTextDecryptionMode) {
+					cstb.btnDecryption.setEnabled(true);
+				}
+				else {
+					if(cstb.txtCiphertextSegments.getText().isEmpty())
+						cstb.btnDecryption.setEnabled(true);
+					else
+						cstb.btnDecryption.setEnabled(false);
+				}
+			}
+			else {
+				cstb.btnDecryption.setEnabled(false);
+			}
+		}
+		
+		
+		/*if(cstb.btnRadioDecrypt.getSelection()) {
+			cstb.txtCiphertextSegments.setText("");
+			if(!cstb.rftc.txtModN.getText().isEmpty() && cstb.textSelector.getText() != null && !cstb.textSelector.getText().getText().isEmpty()) {
+				String pattern = "[0-9a-fA-F]+";
+				
+				if(!cstb.textSelector.getText().getText().matches(pattern)) {
+					cstb.btnDecryption.setEnabled(false);
+					return;
+				}
+				
+				String newTextselectorText = cstb.textSelector.getText().getText();
+				if(newTextselectorText != oldTextselectorTextDecryptionMode) {
+					cstb.btnDecryption.setEnabled(true);
+				}
+				else {
+					if(cstb.txtCiphertextSegments.getText().isEmpty())
+						cstb.btnDecryption.setEnabled(true);
+					else
+						cstb.btnDecryption.setEnabled(false);
+				}
+			}
+			else {
+				cstb.btnDecryption.setEnabled(false);
+			}
+				
+		}*/
+	}
+	
+	
+	public void textSelectorAction(CryptosystemTextbookComposite cstb) {
+		if(cstb.btnRadioEncrypt.getSelection()) {
+			
+			if(cstb.textSelector.getText() != null) {
+				this.hideControl(cstb.txtLoadTextWarning);
+				
+				String newTextselectorText = null;
+				
+				// maybe uncomment again, atm its just testing
+				/*if(firstTextselectorAccessEncryptionMode) {
+					oldTextselectorTextEncryptionMode = cstb.textSelector.getText().getText();
+					firstTextselectorAccessEncryptionMode = true;
+					return;
+				}*/
```

Comment:
And that's code which survives for decades in repositories^^
In 99% of the time it's removable.